### ISSUE TITLE
Regressed Issue: buyer identity attach not working

### ISSUE THREAD

**Issue by mbreithu-wtd** [Shopify/checkout-sheet-kit-swift#208]
### What area is the issue related to?

Checkout Sheet Kit

### What version of Checkout Sheet Kit are you using?

1.x 2.x and 3.x

### Steps to Reproduce

Using buy sdk:
Create a new cart
Add product
Attach buyer identity via access token
* can verify via buy sdk that identity is attached
using this library:
open checkout vwith checkoutURL

### Expected Behavior

User should be attached to cart in checkout

### Actual Behavior

Checkout fields are prefilled but no actual user is attached/logged in.
Same result when copying the checkout url in iPhone safari browser.
BUT, working as expected when checkout url is copied into a MacBook safari browser

Similar thing already happened earlier this year: see issue: [https://github.com/Shopify/checkout-sheet-kit-swift/issues/160](url)

This bug has a big impact on us, as this affects our loyalty program, and we need urgent resolution

### Storefront domain

waterdrop-official.myshopify.com

but happens on multiple of our stores

**Comment by QinweiD** [comment]
The same issue, have you resolved this problem?

**Comment by mbreithu-wtd** [comment]
No resolution yet. which is frustrating, as this affects a lot of our users at the moment.
what we know so far is that the attaching via access token seems to work correctly, as the customer can be queried via the carts attached buyer identity. It seems to be a bug in the mobile version of the checkout as everything works when opening the checkout url on a desktop computer, but not when opening on a phone browser or this library (which essentially just opens it in a webview). Weirdly enough, once the checkout is opened on a desktop, the identity seems to attach correctly, as it then works in the checkout sheet as well (for this single cart)

**Comment by markmur** [comment]
@mbreithu-wtd some changes were reverted in the Cart API relating to this issue. Are you still experiencing the regression?

**Comment by mbreithu-wtd** [comment]
@markmur yes the rollback fixed the issue for us. Hoping that it won't suddenly start happening again
